### PR TITLE
Epmwl 1.1 figure out how to inject h2 into project

### PR DESCRIPTION
**Pull request by MrToady**

### CODE REVIEW

File: src/test/java/com/epam/wl/dao/OrderDAOTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Test
+    void testGetAll(){
+        List<Order> expectedOrderList = new ArrayList<>();
+        expectedOrderList.add(new Order(1,1,1, OrderStatus.IN_PROGRESS));
+        expectedOrderList.add(new Order(2,2,2, OrderStatus.NEW));
+        expectedOrderList.add(new Order(3,3,3, OrderStatus.NEW));
+        expectedOrderList.add(new Order(4,4,4, OrderStatus.NEW));
+
+        assertIterableEquals(expectedOrderList, orderDAO.getAll());
```
Comment (TimWinnicheck) [inline src/test/java/com/epam/wl/dao/OrderDAOTest.java]: Usually, it is suggested to use assertThat

File: src/main/java/com/epam/wl/dao/OrderDAO.java [hunk 2]
```diff
@@ -3,18 +3,84 @@
...
+            return ordersList;
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public List<Order> getNew() {
+        return null;
+    }
```
Comment (TimWinnicheck) [inline src/main/java/com/epam/wl/dao/OrderDAO.java]: PreparedStatements in this file can be final

File: src/main/java/com/epam/wl/dao/OrderDAO.java [hunk 3]
```diff
@@ -3,18 +3,84 @@
...
+                    "INSERT INTO order (userid, editionid) VALUES(?, ?);");
+            preparedStatement.setInt(1, user.getId());
+            preparedStatement.setInt(2, book.getId());
+            preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+        }
+    }
 
-    List<Order> getNew();
+    public void setInProgress(int orderID) {
```
Comment (ginsenghillock) [inline src/main/java/com/epam/wl/dao/OrderDAO.java]: I think it reasonable to create one method setStatus that takes orderId and Status

File: src/test/java/com/epam/wl/dao/OrderDAOTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    }
+
+    private DataSource getDataSource() {
+        EmbeddedDatabaseBuilder builder = new EmbeddedDatabaseBuilder();
+        EmbeddedDatabase db = builder
```
Comment (ginsenghillock) [inline src/test/java/com/epam/wl/dao/OrderDAOTest.java]: builder and db variables should be final

File: src/main/java/com/epam/wl/dao/OrderDAO.java [hunk 5]
```diff
@@ -3,18 +3,84 @@
...
+public class OrderDAO {
+    private DataSource dataSource;
 
-    void setInProgress(int orderID);
+    public OrderDAO(DataSource dataSource) {
+        this.dataSource = dataSource;
+    }
 
-    void close(int orderID);
+    public void create(User user, Book book) {
```
Comment (ginsenghillock) [inline src/main/java/com/epam/wl/dao/OrderDAO.java]: input parameters should be final

**Pull request merged**